### PR TITLE
Don't terminate iterate_data when has_next=false

### PR DESCRIPTION
**Pull request by erlendvollset**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/datapoints_subscriptions.py [hunk 1]
```diff
@@ -246,9 +245,8 @@ def iterate_data(
             res = self._post(url_path=self._RESOURCE_PATH + "/data/list", json=body)
             batch = _DatapointSubscriptionBatch._load(res.json())
 
-            yield batch.updates, batch.subscription_changes
-            if not batch.has_next:
-                return
+            yield DatapointSubscriptionBatch(batch.updates, batch.subscription_changes, batch.has_next)
+
```
Comment (haakonvt) [inline cognite/client/_api/datapoints_subscriptions.py:249]: Do we need both of these classes `_DatapointSubscriptionBatch` and `DatapointSubscriptionBatch`?

Comment (doctrino) [reply cognite/client/_api/datapoints_subscriptions.py:249]: I guess it makes sense to have both as we do not want to expose the `partitions` to the end user. However, I suggest making these into a hierarchy: `_DatapointSubscriptionBatchInternal(DatapointSubscriptionBatch)`.

**Review by doctrino — COMMENTED** [review]
Sometimes it is best to get things done yourself 😝

**Review by doctrino — APPROVED** [review]
Looks good to me

**Pull request merged**